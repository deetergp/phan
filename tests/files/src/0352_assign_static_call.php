<?php
class Container352 implements \ArrayAccess
{
    private $values = [];

    public function offsetSet($id, $value) {
        $this->values[$id] = $value;
    }

    public function offsetGet($id) {}

    public function offsetExists($id) {}

    public function offsetUnset($id) {}
}

class Registry352
{
   private static $data = null;

   public static function getData() {
       if (self::$data === null) {
           self::$data = new Container352;
       }
       return self::$data;
   }
}

function test352() {
    $reg = new Registry352;
    $reg::getData()['foo'] = 'bar';
    $reg->getData()['x'] = 'y';
}
test352();
