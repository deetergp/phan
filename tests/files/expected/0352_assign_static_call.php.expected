%s:10 PhanTypeMissingReturn Method \Container352::offsetGet is declared to return mixed but has no return value
%s:12 PhanTypeMissingReturn Method \Container352::offsetExists is declared to return bool but has no return value
